jobs/payment-jobs/tasks/ejv_partner_distribution_task.py

```diff
@@ -54,230 +56,226 @@ def create_ejv_file(cls):
             .filter(InvoiceModel.disbursement_status_code == DisbursementStatus.COMPLETED.value) \
-            .filter(InvoiceModel.corp_type_code == partner.code) \
             .all()
-        current_app.logger.info(invoices)
-        return invoices
+
+        disbursement_rows = []
+        for invoice, payment_line_item, distribution_code in transactions + reversals:
+            disbursement_rows.append(Disbursement(
+                bcreg_distribution_code=distribution_code,
+                partner_distribution_code=distribution_code.disbursement_distribution_code,
+                line_item=DisbursementLineItem(
+                    amount=payment_line_item.total,
+                    flow_through=f'{invoice.id:<110}',
+                    description_identifier=f'#{invoice.id}',
+                    is_reversal=invoice.invoice_status_code in [InvoiceStatus.REFUNDED.value,
+                                                                InvoiceStatus.REFUND_REQUESTED.value,
+                                                                InvoiceStatus.CREDITED.value],
+                    target_type=EJVLinkType.INVOICE.value,
+                    identifier=invoice.id,
+                    is_legacy=True
+                )
+            ))
+        # ################################################################# END OF Legacy way of handling disbursements.
+        # Partner disbursements - New
+        # Partial refunds need to be added to here later, although they should be fairly rare as most of them are from
+        # NRO (NRO is internal, meaning no disbursement needed.)
+        partner_disbursements = db.session.query(PartnerDisbursementsModel,
+                                                 PaymentLineItemModel,
+                                                 DistributionCodeModel) \
+            .join(PaymentLineItemModel, and_(PaymentLineItemModel.invoice_id == PartnerDisbursementsModel.target_id,
+                  PartnerDisbursementsModel.target_type == EJVLinkType.INVOICE.value)) \
+            .join(DistributionCodeModel,
+                  DistributionCodeModel.distribution_code_id == PaymentLineItemModel.fee_distribution_id) \
+            .filter(PartnerDisbursementsModel.status_code == DisbursementStatus.WAITING_FOR_JOB.value) \
+            .filter(PartnerDisbursementsModel.partner_code == partner.code) \
+            .filter(DistributionCodeModel.stop_ejv.is_(False) | DistributionCodeModel.stop_ejv.is_(None)) \
+            .filter(~InvoiceModel.receipts.any(cast(ReceiptModel.receipt_date, Date) >= disbursement_date.date())) \
+            .order_by(DistributionCodeModel.distribution_code_id, PaymentLineItemModel.id) \
+            .all()
+
+        for partner_disbursement, payment_line_item, distribution_code in partner_disbursements:
+            suffix = 'PR' if partner_disbursement.target_type == EJVLinkType.PARTIAL_REFUND else ''
```

Comment:
EJVLinkType.PARTIAL_REFUND.value?

Reply:
Not implemented, will implement partial refunds in the ticket